src/ssl.c

```diff
@@ -5614,20 +5615,21 @@ int ProcessBuffer(WOLFSSL_CTX* ctx, const unsigned char* buff,
             if (ret == 0) {
                 XMEMCPY(der->buffer, buff, length);
             }
-
-        #ifdef HAVE_PKCS8
-            /* if private key try and remove PKCS8 header */
-            if (type == PRIVATEKEY_TYPE) {
-                if ((ret = ToTraditional_ex(der->buffer, der->length, &algId)) > 0) {
-                    /* Found PKCS8 header */
-                    /* ToTraditional_ex moves buff and returns adjusted length */
-                    der->length = ret;
-                }
-                ret = 0; /* failures should be ignored */
-            }
-        #endif
         }
     }
+    
+    #ifdef HAVE_PKCS8
+    /* if private key try and remove PKCS8 header */
+    if (ret == 0 && type == PRIVATEKEY_TYPE) {
+        if ((ret = ToTraditional_ex2(der->buffer, der->length,
+                                                    &algId, &crvId, 1)) > 0) {
+                /* Found PKCS8 header */
+                /* ToTraditional_ex moves buff and returns adjusted length */
```

Comment:
`ToTraditional_ex` -> `ToTraditional_ex2` in the comment.

Reply:
thanks! Fixed